packages/core/src/accessanalyzer/vue/iamPolicyChecks.ts

```diff
@@ -71,12 +106,266 @@ export class IamPolicyChecksWebview extends VueWebview {
+                                        )
+                                    }
+                                })
+                                this.onValidatePolicyResponse.fire([
+                                    'Please view the problems panel to review the issues with your policy document. Policy checks should be run until issues are no longer found in your policy document.',
+                                    'yellow',
+                                ])
+                                void vscode.commands.executeCommand('workbench.actions.view.problems')
+                            } else {
+                                this.onValidatePolicyResponse.fire([
+                                    'Policy checks did not discover any problems with your policy.',
+                                    'green',
+                                ])
+                            }
+                        }
+                    }
+                )
+                return
+            }
+            case PolicyChecksDocumentType.TERRAFORM_PLAN: {
+                const tfCommand = `tf-policy-validator validate --template-path ${document} --region ${this.region} --config ${defaultTerraformConfigPath}`
+                this.executeValidatePolicyCommand(tfCommand)
+                return
+            }
+            case PolicyChecksDocumentType.CLOUDFORMATION: {
+                const cfnCommand =
+                    cfnParameterPath === ''
+                        ? `cfn-policy-validator validate --template-path ${document} --region ${this.region}`
+                        : `cfn-policy-validator validate --template-path ${document} --region ${this.region} --template-configuration-file ${cfnParameterPath}`
+                this.executeValidatePolicyCommand(cfnCommand)
+                return
+            }
+        }
+    }
+
+    public async checkNoNewAccess(
+        documentType: string,
+        policyType: string,
+        referenceDocument: string,
+        cfnParameterPath?: string
+    ) {
+        const document = IamPolicyChecksWebview.editedDocumentFileName
+        customPolicyCheckDiagnosticCollection.clear()
+        if (referenceDocument !== '') {
+            fs.writeFileSync(tempFilePath, referenceDocument)
+        } else {
+            this.onCustomPolicyCheckResponse.fire(['Reference policy document is missing.', 'red'])
+            return
+        }
+
+        switch (documentType) {
+            case PolicyChecksDocumentType.TERRAFORM_PLAN: {
+                const tfCommand = `tf-policy-validator check-no-new-access --template-path ${document} --region ${this.region} --config ${defaultTerraformConfigPath} --reference-policy ${tempFilePath} --reference-policy-type ${policyType}`
+                this.executeCustomPolicyChecksCommand(tfCommand)
+                return
+            }
+            case PolicyChecksDocumentType.CLOUDFORMATION: {
+                const cfnCommand =
```

Comment:
better to make these logic a helper method so it's unit testable and easier to extend in future.

Reply:
Will move it in the future. Adding to ticket.